bridge/contracts/Bridge/Bridge.sol

```diff
@@ -124,47 +125,45 @@ contract Bridge is Initializable, IBridge, IERC777Recipient, UpgradablePausable,
 		return chainId == block.chainid;
 	}
 
-	function sideTokenAddressByOriginalTokenAddress(uint256 chainId, address originalToken) public view returns(address) {
-		// specification for retrocompatibility
-		if (isCurrentChainId(chainId)) {
-			address sideToken = deprecatedMappedTokens[originalToken];
-			if (sideToken != NULL_ADDRESS) {
-				return sideToken;
-			}
+	function sideTokenByOriginalToken(uint256 chainId, address originalToken) public view returns(address) {
+		address sideTokenAddr = sideTokenByOriginalTokenByChain[chainId][originalToken];
+
+		if (sideTokenAddr != NULL_ADDRESS) {
+			return sideTokenAddr;
 		}
 
-		return sideTokenAddressByOriginalTokenAddressByChain[chainId][originalToken];
+		// specification for retrocompatibility
+		return deprecatedMappedTokens[originalToken];
 	}
 
 	function setSideTokenByOriginalAddressByChain(uint256 chainId, address originalToken, address sideToken) public onlyOwner {
-		sideTokenAddressByOriginalTokenAddressByChain[chainId][originalToken] = sideToken;
+		sideTokenByOriginalTokenByChain[chainId][originalToken] = sideToken;
 	}
 
-	function originalTokenAddressBySideTokenAddress(uint256 chainId, address sideToken) public view returns(address) {
-		// specification for retrocompatibility
-		if (isCurrentChainId(chainId)) {
-			address originalToken = deprecatedOriginalTokens[sideToken];
-			if (originalToken != NULL_ADDRESS) {
-				return originalToken;
-			}
+	function getOriginalTokenBySideToken(address sideToken) public view returns(OriginalToken memory originalToken) {
+		originalToken = originalTokenBySideToken[sideToken];
+		if (originalToken.tokenAddress != NULL_ADDRESS) {
+			return originalToken;
 		}
 
-		return originalTokenAddressBySideTokenAddressByChain[chainId][sideToken];
+		// specification for retrocompatibility
+		originalToken.originChainId = 1; // ethereum main chain id
```

Comment:
originChainId depends on where this contract was deployed originally, so it can be 42 if deployed on kovan, 31 if deployed on rsk testnet, 1 for ethereum mainnet or 30 if rsk mainnet